src/python/pants/backend/python/tasks2/select_interpreter.py

```diff
@@ -50,22 +55,28 @@ def product_types(cls):
 
   def execute(self):
     interpreter = None
-    python_tgts = self.context.targets(lambda tgt: isinstance(tgt, PythonTarget))
-    fs = PythonInterpreterFingerprintStrategy()
-    with self.invalidated(python_tgts, fingerprint_strategy=fs) as invalidation_check:
-      # If there are no relevant targets, we still go through the motions of selecting
-      # an interpreter, to prevent downstream tasks from having to check for this special case.
-      if invalidation_check.all_vts:
-        target_set_id = VersionedTargetSet.from_versioned_targets(
-            invalidation_check.all_vts).cache_key.hash
-      else:
-        target_set_id = 'no_targets'
-      interpreter_path_file = self._interpreter_path_file(target_set_id)
-      if not os.path.exists(interpreter_path_file):
-        self._create_interpreter_path_file(interpreter_path_file, python_tgts)
-
-    if not interpreter:
-      interpreter = self._get_interpreter(interpreter_path_file)
+
+    fixed_interpreter = self.get_options().fixed_interpreter
+
+    if fixed_interpreter:
+      interpreter = PythonInterpreter.from_binary(fixed_interpreter)
+    else:
+      python_tgts = self.context.targets(lambda tgt: isinstance(tgt, PythonTarget))
```

Comment:
Use GitHub's secret whitespace ignore option to see that I haven't actually changed any of this: https://github.com/pantsbuild/pants/pull/4930/files?w=1

Reply:
Neat! Didn't know that existed. Although I'm not sure what that does exactly, esp. where whitespace is meaningful. 

Anyway, might be better to split this branch off into a separate method, for readability.